mobile-widgets/qmlmanager.cpp

```diff
@@ -36,6 +36,13 @@ QMLManager *QMLManager::m_instance = NULL;
 
 #define NOCLOUD_LOCALSTORAGE format_string("%s/cloudstorage/localrepo[master]", system_default_directory())
 
+extern "C" void showErrorFromC(char *buf)
+{
+	QString error(buf);
+	free(buf);
+	emit QMLManager::instance()->showError(error);
```

Comment:
can be replaced by:
```
QMetaObject::invokeMethod(QMLManager::instance(),
	"showError", Qt::QueuedConnection, Q_ARG(QString, error));
```

Reply:
Not sure if I like this. :( The whole only-runtime checked / duck typed thing makes me uneasy. Mistype `registerError` or rename the function and you will not get a compile-time error, but subtle behavioral changes. There seems to be a version of `invokeMethod` taking pointer-to-member-function, but only in Qt 5.10+.

So we have the choice between `connect`, which is basically COME FROM and `invokeMethod` which is weakly-typed Python-esque scripting.

Both sound equally unsatisfying to me. :( Any preferences?

Reply:
i'm currently contributing to QtConnectivity where we had the exact same "connect-to-self" situation and one of the maintainers proposed and accepted the `invokeMethod` solution. if a method is renamed `invokeMethod` reports and error and/or crashes.